Test workflow runs against services that were deleted in the PR

## Summary

When a PR deletes a service directory, the `test` workflow still tries to run `task -- <service>` for that service and fails. Seen on #178, which deletes `services/mysql-oraclelinux/`:

```
task: services/mysql-oraclelinux does not exist.
task: precondition not met
##[error]Process completed with exit code 201.
```

Job: https://github.com/TugboatQA/images/actions/runs/25971530987/job/76344291692

## Cause

`.github/workflows/detect-pr-services.sh` derives the service list purely from the file paths changed in the PR diff:

```bash
git diff --name-only origin/main | \
    grep -Eo '^services/[^/]+' | \
    sed -e 's@^services/@@g' | \
    sort | \
    uniq | \
    xargs echo
```

`git diff --name-only` includes deleted files, so a removed service's name still ends up in the matrix. The downstream `task -- <service>` fails its existence precondition.

## Proposed fix

After the existing pipeline, filter out names whose `services/<name>/` directory does not exist in the working tree (i.e. the service was deleted in this PR). Then we only ever test services that are present on the PR branch.

--- a/.github/workflows/detect-pr-services.sh
+++ b/.github/workflows/detect-pr-services.sh
@@ -6,4 +6,7 @@ git diff --name-only origin/main | \
     sed -e 's@^services/@@g' | \
     sort | \
     uniq | \
+    while read -r service; do
+        [[ -d "services/$service" ]] && echo "$service"
+    done | \
     xargs echo

